src/components/LocalFileUpload/custom-request-upload.ts

```diff
@@ -62,68 +55,81 @@ export default (
     // and not have to have this action called multiple places in the code.
     clearSimulariumFile({ newFile: true });
 
-    const files: FileHTML[] = Array.from(selectedFiles) as FileHTML[];
+    const filesArr: File[] = Array.from(selectedFiles) as File[];
 
-    Promise.all(files.map((file: FileHTML) => file.text()))
-        .then((parsedFiles: string[]) => {
-            const simulariumFileIndex = findIndex(files, (file) =>
-                file.name.includes(".simularium")
+    try {
+        // Try to identify the simularium file.
+        // Put all the other files as text based geoAssets.
+        const simulariumFileIndex = findIndex(filesArr, (file) =>
+            file.name.includes(".simularium")
+        );
+        if (simulariumFileIndex === -1) {
```

Comment:
@toloudis I took a look at this because of your comment about DRYing out the two `catch` blocks, and saw that you had removed this `throw new Error` if the index was -1. So it seems like the try statement would not error out if there was not a `.simularium` file included, so I added the error back in and wanted you to review this to make sure I hadn't missed something

Reply:
That was inadvertent, probably just pasted some new code in and missed that I deleted this block.  Keeping it in is good.